Update CONTRIBUTING.md

gawi151 opened the pull request:
Updates [CONTRIBUTING.md](https://github.com/nomtek/mistralai_client_dart/blob/22-add-guidelines-how-to-contribute-to-the-project-in-contributingmd/CONTRIBUTING.md) with guidance on how to contribute to our project.

gawi151 committed:
first version of contributing.md

MarcinGruchala commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+Before submitting a bug report, please check if the issue has already been reported in the [issue tracker](https://github.com/nomtek/mistralai_client_dart/issues). If it hasn't, create a new issue with a clear title and description, including steps to reproduce the bug.
+
```
Maybe it is a good idea to add some  Issue templates(.md file), where we will show how bugs should be reported. 
Additionally, we will fulfill one of the best practices from insights-> community standards in our project.

gawi151 replied:
Addressed here https://github.com/nomtek/mistralai_client_dart/pull/25. I'm closing this conversation on your behalf @MarcinGruchala

MarcinGruchala commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+2. Create a new branch for your feature or bug fix.
+3. Make your changes and commit them with clear and concise messages.
+4. Push your changes to your fork.
+5. [Submit a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests), explaining the changes you made.
```
Github also suggests that we should have a Pull request template as a good practice in insights-> community standards

gawi151 replied:
Addressed [here](https://github.com/nomtek/mistralai_client_dart/pull/26). I'm closing this conversation on your behalf @MarcinGruchala

MarcinGruchala reviewed (dismissed):
Look good :)

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+# Contributing to Mistral AI Client for Dart
 
-## Development flow
+Thank you for considering contributing to [Your Project Name]! We appreciate your time and effort.
```
We probably should replace the placeholder with our project name 😁 
```suggestion
Thank you for considering contributing to Mistral AI Client for Dart! We appreciate your time and effort.
```

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+### Reporting Bugs
+
+Before submitting a bug report, please check if the issue has already been reported in the [issue tracker](https://github.com/nomtek/mistralai_client_dart/issues). If it hasn't, create a new issue with a clear title and description, including steps to reproduce the bug.
```
It would be nice to also ask users to label the newly created issue with the `BUG` label 😉

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+### Suggesting Enhancements
+
+If you have an idea for an improvement or a new feature, we'd love to hear about it! Create a new issue in the [issue tracker](https://github.com/nomtek/mistralai_client_dart/issues) with your suggestion, and we can discuss it further.
```
It would be nice to also ask users to label the newly created issue with the `ENHANCEMENT` label 😉

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+### Documentation
+
+Improving documentation is a valuable contribution. If you find areas that need clarification or have ideas for better explanations, please make the necessary updates.
```
> please make the necessary updates

But how should they do it? 🤔 I guess they can't simply edit the code without necessary permissions, right? We should explain to them whether they have to create some issues with documentation fixes or if they can do it themselves somehow 🤔

gawi151 replied:
Thx for the suggestion :)

Sorry, I assumed it would go the same way as standard code change. But, as you asked, it's probably not so obvious. I'll try to add some guidance on how to improve documentation.

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+### Setting Up the Development Environment
+
+- Check [pubspec.yaml](https://github.com/nomtek/mistralai_client_dart/blob/main/pubspec.yaml) for any dependencies and versions required to build project.
```
Do they have to check the file? Can't they simply run `dart pub get` and then it's all good to go? 🤔

gawi151 replied:
I wasn't sure what to specify here. As you say `dart pub get` is probably all you need :)

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+## Style Guide
+
+No particular style guide. We are using [very_good_analysis](https://pub.dev/packages/very_good_analysis) package to keep code consistent.
```
It's probably worth adding a style guide in the end 😉 Could you create an issue to do it later? 🙇

gawi151 replied:
Hmm, not sure how a style guide would look like. Do you have any proposals or good and short examples? 

And https://pub.dev/packages/very_good_analysis covers styling lint rules.

KacperKluka replied:
If we're using the default dart style guide that's being enforced by the `very_good_analysis` then let's just specify it here 😉 To me "No particular style guide" sounds like we don't care about code style and that's not true 😅

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+    - Once your changes are approved, a maintainer will merge your pull request.
+    - The changes will be reflected in the main branch.
+
+5. **Celebrate:**
+    - Congratulations! Your contribution has been successfully merged.
```
Maybe we could also include those contributors in the release notes next to the changes they made? 😉

gawi151 replied:
Yeah, that's a good idea. I'll add it.

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka commented on CONTRIBUTING.md:
```diff
@@ -1,5 +1,74 @@
+
+## Getting Started
+
+### Setting Up the Development Environment
```
I think we should have some instructions here that may be obvious to a Flutter/Dart dev but not so obvious to people who are just starting 😉 So things like: download the repo, launch `pub get`, run the project, run tests; might be useful and worth adding, WDYT? 🤔

gawi151 replied:
Now that you mentioned it - it will be nice for newcomers to see some steps on how to do it.

gawi151 replied:
Updated in https://github.com/nomtek/mistralai_client_dart/pull/24/commits/42354570f194dd5e88f3982f0b824e9e1bbca23b

KacperKluka reviewed (changes requested):
A few suggestions from my side 😁

gawi151 committed:
Update project name placeholder

Co-authored-by: KacperKluka <62378170+KacperKluka@users.noreply.github.com>

gawi151 committed:
Merge branch 'main' into 22-add-guidelines-how-to-contribute-to-the-project-in-contributingmd

gawi151 committed:
Update few sections to be more precise and direct

updated sections:
- reporting bugs
- suggesting enhancements
- setting up the development environment
- style guide
- review process

gawi151 committed:
Merge remote-tracking branch 'origin/22-add-guidelines-how-to-contribute-to-the-project-in-contributingmd' into 22-add-guidelines-how-to-contribute-to-the-project-in-contributingmd

KacperKluka reviewed (approved):
LGTM 👍
